radix-transactions/src/validation/transaction_validator.rs

```diff
@@ -379,12 +467,646 @@ impl TransactionValidator {
+        // * Save the children of an intent into its intent details
+
+        // STEP 2A - Handle children of the transaction intent
+        {
+            let parent_hash = root_intent_hash;
+            let intent_details = &mut root_intent_details;
+            for child_subintent in root_intent_core.children.children.iter() {
+                let child_hash = child_subintent.hash;
+                let (subintent_index, _, subintent_details) = all_subintent_details
+                    .get_full_mut(&child_hash)
+                    .ok_or_else(|| {
+                        SubintentValidationError::ChildSubintentNotIncludedInTransaction(child_hash)
+                    })?;
+                if subintent_details.parent == PLACEHOLDER_PARENT {
+                    subintent_details.parent = parent_hash;
+                } else {
+                    return Err(SubintentValidationError::SubintentHasMultipleParents(
+                        child_hash,
+                    ));
+                }
+                intent_details
+                    .children
+                    .push(SubintentIndex(subintent_index));
+            }
+        }
+
+        // STEP 2B - Handle the children of each subintent
+        for subintent in subintents.iter() {
+            let subintent_hash = subintent.subintent_hash();
+            let parent_hash: IntentHash = subintent_hash.into();
+            let children = &subintent.intent_core.children.children;
+            let mut children_details = Vec::with_capacity(children.len());
+            for child_subintent in children.iter() {
+                let child_hash = child_subintent.hash;
+                let (subintent_index, _, subintent_details) = all_subintent_details
+                    .get_full_mut(&child_hash)
+                    .ok_or_else(|| {
+                        SubintentValidationError::ChildSubintentNotIncludedInTransaction(child_hash)
+                    })?;
+                if subintent_details.parent == PLACEHOLDER_PARENT {
+                    subintent_details.parent = parent_hash;
+                } else {
+                    return Err(SubintentValidationError::SubintentHasMultipleParents(
+                        child_hash,
+                    ));
+                }
+                children_details.push(SubintentIndex(subintent_index));
+            }
+            all_subintent_details
+                .get_mut(&subintent_hash)
+                .unwrap()
+                .children = children_details;
+        }
+
+        // STEP 3
+        // ------
+        // We traverse the child relationships from the root, and mark a depth.
+        // We error if any exceed the maximum depth.
+        //
+        // As each child has at most one parent, we can guarantee the work is bounded
+        // by the total number of subintents.
+        let mut work_list = vec![];
+        for index in root_intent_details.children.iter() {
+            work_list.push((*index, 1));
```

Comment:
depth typically start with `0` at the root.

Reply:
I agree - these are the depths for the root's children (i.e. the first layer of subintents)